src/core/pluginmanager.h

```diff
@@ -134,6 +124,8 @@ class PluginManager : public QObject
     QString mPermissionRequestPluginPath;
 
     QMap<QString, PluginInformation> mAvailableAppPlugins;
+
+    QList<PluginInformation> mPublicPlugins;
```

Comment:
OK, here I think we should not have a list of plugins but instead add a std::unique_ptr<PluginModel> mPluginModel and move all  plugins listing (and remote fetching, etc.) into that model class.